launcher/QObjectPtr.h

```diff
@@ -1,91 +1,48 @@
 #pragma once
 
+#include <QObject>
+#include <QSharedPointer>
 #include <functional>
 #include <memory>
-#include <QObject>
 
-namespace details
-{
-struct DeleteQObjectLater
+namespace details {
+[[maybe_unused]] static void do_delete_later(QObject* obj)
 {
-    void operator()(QObject *obj) const
-    {
+    if (obj)
         obj->deleteLater();
-    }
-};
 }
+struct DeleteQObjectLater {
+    void operator()(QObject* obj) const { do_delete_later(obj); }
+};
+
+}  // namespace details
+
 /**
  * A unique pointer class with unique pointer semantics intended for derivates of QObject
  * Calls deleteLater() instead of destroying the contained object immediately
  */
-template<typename T> using unique_qobject_ptr = std::unique_ptr<T, details::DeleteQObjectLater>;
+template <typename T>
+using unique_qobject_ptr = std::unique_ptr<T, details::DeleteQObjectLater>;
 
 /**
  * A shared pointer class with shared pointer semantics intended for derivates of QObject
  * Calls deleteLater() instead of destroying the contained object immediately
  */
 template <typename T>
-class shared_qobject_ptr
-{
-public:
-    shared_qobject_ptr(){}
-    shared_qobject_ptr(T * wrap)
-    {
-        reset(wrap);
-    }
-    shared_qobject_ptr(const shared_qobject_ptr<T>& other)
-    {
-        m_ptr = other.m_ptr;
-    }
-    template<typename Derived>
-    shared_qobject_ptr(const shared_qobject_ptr<Derived> &other)
-    {
-        m_ptr = other.unwrap();
-    }
+class shared_qobject_ptr : public QSharedPointer<T> {
+   public:
+    constexpr shared_qobject_ptr() : QSharedPointer<T>() {}
+    constexpr shared_qobject_ptr(T* ptr) : QSharedPointer<T>(ptr, details::do_delete_later) {}
+    constexpr shared_qobject_ptr(std::nullptr_t null_ptr) : QSharedPointer<T>(null_ptr, details::do_delete_later) {}
```

Comment:
```
QSharedPointer<MyObject> obj = QSharedPointer<MyObject>(new MyObject, &QObject::deleteLater);
```

^ from Qt docs

Reply:
AFAICT we need the `if (obj)` from `details::do_delete_later` to avoid that warning. Also using `QSharedPointer` directly would require much more changes in the codebase, right? .-.